marshmallow/fields.py

```diff
@@ -401,23 +401,36 @@ def schema(self):
 
         # Inherit context from parent.
         context = getattr(self.parent, 'context', {})
+
+        # Normalize nested load_only and dump_only from parent sets
+        nested_field = '%s.' % self.name
```

Comment:
This implementation has performance impact of calculating this every time schema is requested although schema itself is being cached. I suggest moving this into two functions and call them whenever you create schema class.